src/transformers/models/qwen2_vl/image_processing_qwen2_vl.py

```diff
@@ -0,0 +1,454 @@
+                - Unset: Use the channel dimension format of the input image.
+            input_data_format (`ChannelDimension` or `str`, *optional*):
+                The channel dimension format for the input image. Can be one of:
+                - `"channels_first"` or `ChannelDimension.FIRST`: image in (num_channels, height, width) format.
+                - `"channels_last"` or `ChannelDimension.LAST`: image in (height, width, num_channels) format.
+                - `"none"` or `ChannelDimension.NONE`: image in (height, width) format.   - `"none"` or `ChannelDimension.NONE`: image in (height, width) format.
+        """
+        images = make_list_of_images(images)
+
+        if do_convert_rgb:
+            images = [convert_to_rgb(image) for image in images]
+
+        # All transformations expect numpy arrays.
+        images = [to_numpy_array(image) for image in images]
+
+        if is_scaled_image(images[0]) and do_rescale:
+            logger.warning_once(
+                "It looks like you are trying to rescale already rescaled images. If the input"
+                " images have pixel values between 0 and 1, set `do_rescale=False` to avoid rescaling them again."
+            )
+        if input_data_format is None:
+            # We assume that all images have the same channel dimension format.
+            input_data_format = infer_channel_dimension_format(images[0])
+
+        height, width = get_image_size(images[0], channel_dim=input_data_format)
+        resized_height, resized_width = height, width
+        if do_resize:
+            resized_height, resized_width = smart_resize(
+                height,
+                width,
+                factor=self.patch_size * self.merge_size,
+                min_pixels=self.min_pixels,
+                max_pixels=self.max_pixels,
+            )
+            images = [
+                resize(
+                    image=image,
+                    size=(resized_height, resized_width),
+                    resample=resample,
+                    input_data_format=input_data_format,
+                )
+                for image in images
+            ]
+
+        if do_rescale:
+            images = [
+                self.rescale(image=image, scale=rescale_factor, input_data_format=input_data_format)
+                for image in images
+            ]
+
+        if do_normalize:
+            images = [
+                self.normalize(image=image, mean=image_mean, std=image_std, input_data_format=input_data_format)
+                for image in images
+            ]
+
+        images = [
+            to_channel_dimension_format(image, data_format, input_channel_dim=input_data_format) for image in images
+        ]
+
+        patches = np.array(images)
+        if patches.shape[0] == 1:
+            patches = np.tile(patches, (self.temporal_patch_size, 1, 1, 1))
+        c = patches.shape[1]
+        grid_t = patches.shape[0] // self.temporal_patch_size
```

Comment:
Still not sure that we can assume it is a channel-first image, as `data_format` is easily modifiable by users. Can we make it channel dim agnostic? Also, casting "np.array" isn't needed, all images should be arrays by this point

Reply:
What if we want the image only be channel-first?

Reply:
Should we give an assert?

Reply:
I am not sure if we can restrict it to be always channel-first, haven't seen such cases yet. @amyeroberts wdyt?

Reply:
All the image transformations should be channel agnostic. You can always force images to be one format for the transformation within a function or method, but the behaviour must be consistent i.e. a channels last image is returned as a channels last image unless `data_format="channels_first"` is specified

Reply:
Our vision encoder does not care about the `data_format` of input images, as we reshape them to patch sequence. So if the `data_foramt="channel_last"`, we just transpose them to `channel_first` before the patch sequence reshaping.